examples/hybrid/bubble-mountain.jl

```diff
@@ -0,0 +1,170 @@
+push!(LOAD_PATH, joinpath(@__DIR__, "..", ".."))
+
+using Test
+using StaticArrays, IntervalSets, LinearAlgebra, UnPack
+
+import ClimaCore:
+    ClimaCore,
+    slab,
+    Spaces,
+    Domains,
+    Meshes,
+    Geometry,
+    Topologies,
+    Spaces,
+    Fields,
+    Operators
+using ClimaCore.Geometry
+
+
+
+
+function warp_agnesi_peak(coord; z_top = 1000.0, a = 1 / 2)
+
+    h = 8 * a^3 / (x_in^2 + 4 * a^2)
+    x, z = x_in, z_in + h
+    return x, z
+end
+
+
+# set up function space
+function hvspace_2D(
+    xlim = (-π, π),
+    zlim = (0, 4π),
+    helem = 10,
+    velem = 50,
+    npoly = 4;
+    stretch = Meshes.NoStretching(),
+    topography_file = nothing,
+)
+
+    # build vertical mesh information with stretching in [0, H]
+    FT = Float64
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(zlim[1]),
+        Geometry.ZPoint{FT}(zlim[2]);
+        boundary_tags = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, stretch, nelems = velem)
+    vert_face_space = Spaces.FaceFiniteDifferenceSpace(vertmesh)
+
+
+
+
+    # build horizontal mesh information
+    horzdomain = Domains.IntervalDomain(
+        Geometry.XPoint{FT}(xlim[1])..Geometry.XPoint{FT}(xlim[2]),
+        periodic = true,
+    )
+    horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
+    horztopology = Topologies.IntervalTopology(horzmesh)
+
+    quad = Spaces.Quadratures.GLL{npoly + 1}()
+    horzspace = Spaces.SpectralElementSpace1D(horztopology, quad)
+
+
+    # todo do we seperate hv_center_space & hv_face_space
+    # construct hv center/face spaces, recompute metric terms
+
+    hv_face_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_face_space)
+end
+
+# set up rhs!
+space = hvspace_2D((-500, 500), (0, 1000))
+
```

Comment:
@simonbyrne  Should we also add stretching here? we have them in interval.jl : for example NoStretching, ExponentialStretching,  GeneralizedExponentialStretching,